os/mm/mm_heap/mm_heapinfo.c

```diff
@@ -217,11 +218,22 @@ void heapinfo_parse(FAR struct mm_heap_s *heap, int mode, pid_t pid)
 	printf("\n****************************************************************\n");
 	printf("     Summary of Heap Usages (Size in Bytes)\n");
 	printf("****************************************************************\n");
-	printf("Total                           : %u (100%%)\n", heap->mm_heapsize);
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	heap_size = heap->mm_heapsize - heap->elf_data_size;
+	if (heap->elf_data_size != 0) {
+		printf("Total                           : %u (100%%)\n", heap_size);
+
+	} else {
+#endif
+		heap_size = heap->mm_heapsize;
+		printf("Total                           : %u (100%%)\n", heap_size);
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	}
+#endif
```

Comment:
How about we write the code as below:

`heap_size = heap->mm_heapsize;`
`#ifdef CONFIG_APP_BINARY_SEPARATION`
`heap_size -= heap->elf_data_size;`
`#endif`
`printf("Total                           : %u (100%%)\n", heap_size);`

Reply:
Done. Thanks.